infra/src/main/kotlin/fi/fta/geoviite/api/frameconverter/v1/FrameConverterServiceV1.kt

```diff
@@ -97,41 +141,27 @@ class FrameConverterServiceV1 @Autowired constructor(
             }
         )
 
-        val mappedLocationTrackTypeOrNull = when (request.locationTrackType) {
-            null -> null
-
-            FrameConverterLocationTrackTypeV1.MAIN -> LocationTrackType.MAIN
-            FrameConverterLocationTrackTypeV1.SIDE -> LocationTrackType.SIDE
-            FrameConverterLocationTrackTypeV1.CHORD -> LocationTrackType.CHORD
-            FrameConverterLocationTrackTypeV1.TRAP -> LocationTrackType.TRAP
-
-            else -> {
-                errors.add("invalid-location-track-type")
-                null
+        val mappedLocationTrackTypeOrNull = mapLocationTrackTypeToDomainTypeOrNull(request.locationTrackType)
+            .let { (mappedType, errorKey) ->
+                errorKey?.let { errors.add(errorKey) }
```

Comment:
Nämä letit voi myös tehdä muodossa `?.let(errors::add)`. Ei sinällään muuta mitään. En itse jotenkin tykkää tuosta että käytetään lambdan ulkopuolista muuttujaa lambdan sisällä, vaikka se tällaisessa paikassa kyllä onkin aivan sama kuin käyttää `it` tai nimetä se uudelleen lambdassa. Letin sijasta voisi olla myös semanttisesti oikeaoppisempaa käyttää also-funkkaria koska paluuarvolla ei tehdä mitään. Tietysti olisi vielä kivompi jos keksisi rakenteen jossa ei tarvitse tehdä tuota listan mutatointia ollenkaan, mutta ei tämä nyt mitenkään liian pahalta näytä.